core/src/main/cpp/io/CMakeLists.txt

```diff
@@ -2,6 +2,12 @@ message(STATUS "io")
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DRAPIDJSON_HAS_STDSTRING=1")
 
+# Ignore new warning for conversion from 64 to 32 bits, we should
+# not have a MSL payload larger than 2^32 bytes.
+set_source_files_properties(BufferedInputStream.cpp PROPERTIES COMPILE_FLAGS "-Wno-shorten-64-to-32")
```

Comment:
What's the source of the problem here?

Reply:
There are places where we the return type is an integer, but the values returned by the method are size_t.  I've removed this, and have chosen to cast to int instead.